### DISCUSSION TITLE
Comparison line on the main graph

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#532] [Anything Else]
"For the graph, would be cool to have a comparison line (previous period) in a different color and lower opacity like Stripe do"

([feature request via Twitter](https://twitter.com/VictorBaudot_/status/1343992791421112321))

Example from GA:

![2021-01-02-091032_grim](https://user-images.githubusercontent.com/34340819/103453484-cfee1400-4cda-11eb-8089-64e2b7082373.png)

**Update:** You can now compare your stats over time! Select "**Compare**" in the date picker or press the "**X**" key on your keyboard to start comparing. Learn more about the comparisons here: https://plausible.io/docs/compare-stats.

**Comment by davidkarpik** [comment]
Oh definitely, I'd love to see that.

**Comment by Vigasaurus** [comment]
So here's my first pass at this:
![image](https://user-images.githubusercontent.com/19434157/105005874-7e909500-59fb-11eb-8179-97aef8d3d2ca.png)
![image](https://user-images.githubusercontent.com/19434157/105005881-805a5880-59fb-11eb-8063-4a444ca01a42.png)

The tooltip formatting isn't quite done yet, but I'd love any feedback on the main idea/the wording on the tooltip (the title feels a bit wordy to me for how big it is, we could bring down its size or have a more concise way to convey the difference). Also from what I'm seeing it's not that easy to do text coloring in the tooltip for the percent change, so if there's a better way to convey that without color/only with text that anyone can think of, let me know.

**Reply by davidkarpik** [reply]
The graph looks great already, maybe it would make sense to make dashed line for the previous period? The difference in the tooltip could be shown with (color coded) arrows (up/down) + percentage and visitor amount, example: 270 Visitors ↓10%  ↓30V.

**Reply by Vigasaurus** [reply]
The only reason I went for opacity change is because we're already using dashed for the "current time period" of the graph - i.e. the last link on the present graph to indicate the period isn't complete yet. I do like the idea of showing the actual change in number too, but I worry it'll make the graph a bit more complicated than it needs to be - I think most people do really care a lot more about the percent change. 

As for color coding the arrows, that is the goal - but it's seeming to be not that simple with the current graph library we use - so I might try to make that change, but ultimately that's why I was looking for a potential other solution that could be done with just text (e.g. emojis - something like 📈 and 📉 or 🔻 and 🔼 but those are inconsistent in color and style on various OSes at times).

**Reply by metmarkosaric** [reply]
looks great @Vigasaurus! here's an example of a tooltip that I think is designed nicely and which makes a lot of sense. what do you think?

![Screenshot from 2021-01-19 10-07-39](https://user-images.githubusercontent.com/34340819/105012271-45c5df80-5a3e-11eb-90b0-f2972b975a61.png)

**Reply by Vigasaurus** [reply]
Huh yeah I quite like that, both for formatting and data clarity. It does a good job of making it clear what's being compared and the dates.

I'll see what I can do 👍 the library does support fully custom tooltips and that seems to be the only way to get meaningful formatting there, so I'll try that out and post new screenshots as appropriate. 

How do you think the hourl-based difference should be conveyed in this format? Just show something like `Jan 18, 9pm` and `Jan 19, 9pm`? Or just the dates?

**Reply by metmarkosaric** [reply]
nice! yeah i think like that with hours would be fine unless you feel it makes it too busy. and thanks again for your help @Vigasaurus !

**Reply by Vigasaurus** [reply]
Second Pass - thoughts? @metmarkosaric 

![image](https://user-images.githubusercontent.com/19434157/105126696-f070e900-5aa4-11eb-932f-64b0ae4f70da.png)
![image](https://user-images.githubusercontent.com/19434157/105126504-81939000-5aa4-11eb-9e52-e408c9ba8d6d.png)
![image](https://user-images.githubusercontent.com/19434157/105126746-09799a00-5aa5-11eb-80fb-597321ab6468.png)
![image](https://user-images.githubusercontent.com/19434157/105128065-d7b60280-5aa7-11eb-9a3d-ae23f9de2a5e.png)

**Reply by metmarkosaric** [reply]
Looks beautiful @Vigasaurus! Thank you! What do you think about making the top line even more simple? 

Metric name and percentage only. 

No plus/minus signs, no arrow (color is enough to show if it's trending up or down), no brackets and no total number difference (percentage difference should be enough when reporting and you can look at the total numbers below if you really care about the total difference) . I would also remove the decimal points from the percentage so it's just 90% rather than 90.4%. How does that sound?

**Reply by Vigasaurus** [reply]
Yeah that's very nice and clean

Should be PRing soon 🎉 just need to do some code cleanup etc

![image](https://user-images.githubusercontent.com/19434157/105136918-1784e600-5ab8-11eb-90cd-54fc7eb301f3.png)

**Reply by metmarkosaric** [reply]
looks great! looking forward to this!

**Comment by benjaminhouy** [comment]
Looking forward to this. This would be super helpful :)

**Comment by mlent** [comment]
Would be incredibly helpful! Especially for keeping an eye on growth, comparing same days of the week, and spotting broader ranking changes :)

**Reply by Cdingram** [reply]
+1 as well!

**Comment by davidkarpik** [comment]
Any news here? The comparison line is the last thing left that makes me miss analytics. ;(

**Comment by metmarkosaric** [comment]
**Update:** You can now compare your stats over time! Select "**Compare**" in the date picker or press the "**X**" key on your keyboard to start comparing. Learn more about the comparisons here: https://plausible.io/docs/compare-stats.

**Reply by davidkarpik** [reply]
Looks fantastic! Finally :) Thanks.

**Reply by davidkarpik** [reply]
Looks like there's something wrong with the unique visitors and unique conversions, see attached screenshot. The unique conversions and the unique visitors of the previous period in comparison view are always the same.![IMG_20230429_081615.jpg](https://user-images.githubusercontent.com/75309901/235287176-c8cf35bd-bfe7-4535-93b2-40f4c4144667.jpg)

**Reply by metmarkosaric** [reply]
thanks for reporting @davidkarpik! looks like a bug! i've reported it to our devs so we can have a look

**Reply by vinibrsl** [reply]
Hey @davidkarpik! We've just deployed a fix for this bug. Could you please check if it's working for you? Thanks for reporting!

**Reply by davidkarpik** [reply]
It's working now, thanks everyone!
